### PR TITLE
Add some documentation about Rendezvous

### PR DESCRIPTION
**Pull request by vivien-apple**
#### Problem
This PR add a bit of documentation to Rendezvous using the @dot command from Doxygen.

### CODE REVIEW

File: src/transport/RendezvousSession.h [hunk 1]
```diff
@@ -25,6 +31,28 @@
 
 namespace chip {
 
+/**
+ * RendezvousSession establish and maintain the first connection between
```
Comment (andy31415) [inline src/transport/RendezvousSession.h:35]: nit: "establishes and maintains" or "is used to establish and maintain"

**Review by andy31415 — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]
approving pending @andy31415 's comments

File: src/transport/RendezvousSession.h (hunk 1 above)
Comment (vivien-apple) [reply src/transport/RendezvousSession.h]: Done.

File: src/transport/RendezvousSession.h [hunk 2]
```diff
@@ -25,6 +31,28 @@
...
 namespace chip {
 
+/**
+ * RendezvousSession establishs and maintains the first connection between
+ * a commissioner and a device. This connection is used in order to
+ * provide the necessary infos for a device to participate to the CHIP
+ * ecosystem.
+ *
+ * All the informations transmitted over the underlying transport are
+ * encrypted upon estasblishment of an initial secure pairing session.
```
Comment (mspang) [inline src/transport/RendezvousSession.h:41]: estasblishment -> establishment

File: src/transport/RendezvousSession.h [hunk 3]
```diff
@@ -25,6 +31,28 @@
...
+ *
+ * In order to securely transmit the informations, RendezvousSession
```
Comment (mspang) [inline src/transport/RendezvousSession.h:43]: informations -> information

File: src/transport/RendezvousSession.h [hunk 4]
```diff
@@ -25,6 +31,28 @@
 
 namespace chip {
 
+/**
+ * RendezvousSession establishs and maintains the first connection between
```
Comment (mspang) [inline src/transport/RendezvousSession.h:35]: establishs -> establishes

Comment (vivien-apple) [reply src/transport/RendezvousSession.h]: Done.

File: src/transport/RendezvousSession.h (hunk 3 above)
Comment (vivien-apple) [reply src/transport/RendezvousSession.h:43]: Done.

File: src/transport/RendezvousSession.h (hunk 2 above)
Comment (vivien-apple) [reply src/transport/RendezvousSession.h]: Done.

**Review by BroderickCarlin — APPROVED** [review]

**Pull request merged**